### DISCUSSION TITLE
Collection_order: custom setting

### DISCUSSION THREAD

**Discussion by 2fst4u** [meisnate12/Plex-Meta-Manager#410] [Q&A]
When using this setting, should the collection be ordered as it appears in the list, when using IMDb for example? I have tried using this and it seems to just build the collection in the release date order.

My movies.yml looks like this:

```
collections:
  IMDb Top 250:
    imdb_list: https://www.imdb.com/list/ls068082370/
    sync_mode: sync
    collection_order: custom
```

**Comment by meisnate12** [comment]
yes it should be in that order

**Reply by dannyvfilms** [reply]
Just added that line in that specific order and started seeing results. Did I miss this item on the Wiki? I had to sort through discussions and issues to find this.

**Reply by meisnate12** [reply]
its under [Collection Details](https://github.com/meisnate12/Plex-Meta-Manager/wiki/Collection-Details#metadata-details) but its not a very big section and it doesn't stand out very well

**Comment by 2fst4u** [comment]
Does anything seem wrong in my syntax? I can't figure out why my collection only builds in release order. I've tried deleting it and recreating it.

**Reply by meisnate12** [reply]
are you using the latest version of PMS and are you using the develop branch or the main branch of PMM

**Reply by 2fst4u** [reply]
PMS is 1.24.4.5081 and I was using the main branch. Should I try develop?

**Reply by 2fst4u** [reply]
Develop did the same thing. I also tried removing the tag from all movies, deleting the collection and trying again and it seems it is now set to "custom" but the order is still release date when I look at them.

The sort order of the movies on the page isn't in release date. What could I be getting wrong?

**Reply by meisnate12** [reply]
can you send me your full meta.log

**Reply by 2fst4u** [reply]
Yea sure what's the best way to send it privately?

**Reply by 2fst4u** [reply]
I'm starting to think there might be something weird going on in my setup. My collection is called "IMDb Top 250" (uppercase T) but the collection created is "IMDb top 250" (lowercase t). I have also tried adding the summary and poster attributes but these aren't showing at all.

**Reply by meisnate12** [reply]
can you hop on discord and send me the log there the PMM invite link is in the main about

**Comment by DrLeoSpaceman** [comment]
Any chance there was a solution to this issue? I'm having the same issue. An IMDB list generated from a search URL made to list by descending IMDB user rating, sort order set to custom in PMM. PMM logs showing it's setting the sort to custom, and in Plex details the collection shows "Custom". However, everything is sorted in alphabetical order.

I'm on 1.12.2-develop1115.

**Reply by meisnate12** [reply]
are you letting the run finish? PMM sorts collections after all collections are run, not when its finished adding to one particular collection. I have it this way because these a bug in plex so i have to do all the sorting at once

**Reply by DrLeoSpaceman** [reply]
Ok, that explains it. Once it got through an entire run everything is now in the proper order. Sorry for being impatient.

(Awesome program, btw.)

**Comment by thunderstorm654** [comment]
It's not sorting collections properly for me either using collection order custom, used to work fine, not sure when it stopped working exactly though as only just noticed. 

PMM run until completion. Its sorting some of them right but others not - interestingly movies is working okay but my TV show ones aren't - no obvious difference in config, hasn't been changed since it used to work etc.

**Comment by Dvx84** [comment]
piling on on this as I'm struggling with it myself and the documentation in the wiki is very unintuitive (love everything else, especially all the work and the quality of the script though).

I'm no python expert and a bit confused. For instance, I am using the proposed code for "Walt Disney Animations", but the order is completely random. 

The template used is Studio with following code: 

Studio:
    optional:
      - company
    tmdb_company: <<company>>
    sort_title: +2_<<collection_name>>
    sync_mode: sync
    smart_label: release.desc

How do I get the script to sort by release date? 

Thanks a ton

**Reply by meisnate12** [reply]
This is a slightly loaded question but I'm gonna try my best to explain as much as possible.
First, I want to note is that its not python you're editing its YAML Files you just run the script using python you don't actually write a single line of python.
Second, the order doesn't matter what matters is the spacing and the grouping its under and its either there or it isn't. 
Third, so how you sort collections depends on if you want the collection to be a smart collection or not.
* Smart Collections: adding `smart_label` and giving it a sort from [this table](https://github.com/meisnate12/Plex-Meta-Manager/wiki/Smart-Builders#sort-options) will sort it in that order.
* Normal Collections: you can use `collection_order` to sort the collection by either `alpha`, `release`, `custom` (which will use the builder list order), or any of the sort listed on [this table](https://github.com/meisnate12/Plex-Meta-Manager/wiki/Smart-Builders#sort-options).

**Reply by Dvx84** [reply]
Hey man,

Thanks again for the super fast reply. So I finally understood the difference between classical collection and smart collection and moved everything to smart collection as it seems to be the better solution from what I read in the wiki (let me know if not ;-) ).

However, I can seem to used the sort function right. I used the sort_by: with the flag release.asc but the order in the collections is completely random. 

Any idea why that could be?

Thanks again for all the great work and support.

**Reply by YozoraXCII** [reply]
If you are using a smart collection, you would use "smart_label: release.asc" rather than "sort_by: release.asc"

**Reply by Dvx84** [reply]
That fixed it. it seems adding sort_by created an error "attribute not recognized" and stopped the collection update. Just noticed that in the logs. All works fine now.

I am just struggling with one last thing on my IMDB top 250. I want to sort them from highest IMDB score to lowest. I know that plex has troubles handling imdb scores also. What is the attribute I need to use to get the sorting right? critic.rating and random don't work. Is there any parameter I can use to get this right?

Thanks again to everyone for the super helpful feedback

**Reply by mark-butterworth-lookinglocal** [reply]
> This is a slightly loaded question but I'm gonna try my best to explain as much as possible.
> First, I want to note is that its not python you're editing its YAML Files you just run the script using python you don't actually write a single line of python.
> Second, the order doesn't matter what matters is the spacing and the grouping its under and its either there or it isn't. 
> Third, so how you sort collections depends on if you want the collection to be a smart collection or not.
> * Smart Collections: adding `smart_label` and giving it a sort from [this table](https://github.com/meisnate12/Plex-Meta-Manager/wiki/Smart-Builders#sort-options) will sort it in that order.
> * Normal Collections: you can use `collection_order` to sort the collection by either `alpha`, `release`, `custom` (which will use the builder list order), or any of the sort listed on [this table](https://github.com/meisnate12/Plex-Meta-Manager/wiki/Smart-Builders#sort-options).

Table links no longer appear to work and I haven't worked out the distinction between smart and normal yet to know which I need. I guess smart as I'm using a radarr builder.
I want order by added.desc and cannot do that with radarr so smart and custom won't work.
So, I need smart_label... Time to dive back in to the wiki.

FYI, amazing application doing some very powerful things. As a software developer I try to congratulate other devs on a job well done.
I personally find it hard to combine branches of the wiki to work out the entire config that meets my needs but that's a common docs issue.

**Comment by echophex** [comment]
Was there ever a sure fire solution for this? Its not sorting correct for me. Here is the suspect collection info in my movies yml:
  Latest releases:
    template: {name: Special Genre}
    sort_title: ++++++++_
    trakt_list: https://trakt.tv/users/garycrawfordgc/lists/latest-releases?sort=added,asc
    collection_order: custom
    summary: "Latest releases"

There is no rhyme or reason for how its actually ordering them.

**Reply by YozoraXCII** [reply]
> 

Are you letting your run finish completely? The reordering won't happen until the end of the run.

You can also try deleting the collection and letting PMM recreate it - make sure the run has finished

I use the same list and I personally replaced collection_order: custom with smart_label: added.desc, that way you're getting things that are new to your Plex instance first in the list.

**Reply by echophex** [reply]
Yes it is finishing but still not listing  the collection in order that the Trakt list is for whatever reason. Appreciate your help tho

**Reply by echophex** [reply]
I also tried taking your suggestion and replacing collection_order: custom with smart_label: added.desc and got this error:
| Latest releases           |  0  |  0  |  0  | PMM Failure                                          |
| Collection Error: collection_order does not work with Smart Collections

This is my exact yml text:
  Latest releases:
    template: {name: Special Genre}
    sort_title: ++++++++++_
    trakt_list: https://trakt.tv/users/giladg/lists/latest-releases?sort=added,asc
    smart_label: release.desc
    summary: "Latest releases"

Mind copying over your exact working yml for your latest releases?

**Reply by meisnate12** [reply]
You're probably using smart_label in the template which turns it into a smart collection which is also why custom ordering wasn't working

**Reply by echophex** [reply]
Meisnate12, you are 1000% correct. Thank you for pointing this out. Very much appreciate your time in responding AND for creating this. Its fantastic
